File: backend/app/api/routes/github_status.py
from fastapi import APIRouter
from app.integrations.github.status_client import (
    get_commit_status,
    get_check_runs
)

router = APIRouter(prefix="/github/status", tags=["github"])

@router.get("/{owner}/{repo}/{sha}")
def get_status(owner: str, repo: str, sha: str, token: str):
    status = get_commit_status(owner, repo, sha, token)
    checks = get_check_runs(owner, repo, sha, token)

    return {
        "state": status.get("state"),
        "statuses": status.get("statuses", []),
        "check_runs": checks.get("check_runs", [])
    }

File: backend/src/api/routes/github-status.ts
import { Router, Request, Response } from 'express';
import { getCommitStatus, getCheckRuns } from '../../integrations/github/status-client.js';
import { GITHUB_DOTCOM_HOST, apiBaseUrlForHost, isAllowedHost } from '../../integrations/github/host.js';

const router = Router();

router.get('/github/status/:owner/:repo/:sha', async (req: Request, res: Response) => {
  try {
    const { owner, repo, sha } = req.params;
    const token = req.query.token as string;
    const host = ((req.query.host as string) || GITHUB_DOTCOM_HOST).toLowerCase();

    if (!token) {
      res.status(400).json({ detail: 'Missing token query parameter' });
      return;
    }
    // The caller's token is forwarded to this host, so it must be allow-listed.
    if (!isAllowedHost(host)) {
      res.status(400).json({ detail: `GitHub host '${host}' is not allowed` });
      return;
    }

    const apiBaseUrl = apiBaseUrlForHost(host);
    const status = await getCommitStatus(owner, repo, sha, token, apiBaseUrl);
    const checks = await getCheckRuns(owner, repo, sha, token, apiBaseUrl);

    res.json({
      host,
      state: (status as Record<string, unknown>).state,
      statuses: (status as Record<string, unknown>).statuses || [],
      check_runs: (checks as Record<string, unknown>).check_runs || [],
    });
  } catch (err) {
    console.error('GitHub status error:', err);
    res.status(500).json({ detail: 'Failed to fetch GitHub status' });
  }
});

export default router;
